### ISSUE TITLE
CI is not using icepack

### ISSUE THREAD

**Issue by travissluka** [JCSDA-internal/soca#968]
icepack is optional, but is enabled in the TravisCI tests ( or at least it was supposed to have been enabled, see https://github.com/JCSDA-internal/soca/pull/967).

It seems icepack is not being built with the new CI. Turn it on.

**Comment by ShastriPaturi** [comment]
@travissluka the release/stable-nightly as of 2023-11-01 fails with:
/work2/noaa/da/spaturi/sandbox/20231101_soca-science/soca-science/bundle/ioda-converters/src/satbias/SatBiasConverter.cpp:65:57: error: no matching function for call to ‘missingValue(float&)’
   65 |   float missing_value = util::missingValue(missing_value);

I added this to the soca bundle:
option(BUILD_ICEPACK "download and build icepack" OFF)
if ( BUILD_ICEPACK )
  ecbuild_bundle( PROJECT icepack         GIT "https://github.com/JCSDA-internal/Icepack.git"         UPDATE BRANCH feature/ecbuild-new )
endif()
and -DBUILD_ICEPACK=ON during ecbuild.

**Comment by shlyaeva** [comment]
This would be nice. Do you think we'd need an update to jedi-bundle and an option in CI to build it, or build it as part of spack-stack?

**Comment by travissluka** [comment]
> This would be nice. Do you think we'd need an update to jedi-bundle and an option in CI to build it, or build it as part of spack-stack?

Yes, the first case: the icepack repo (the jcsda fork of it) should be added to the jedi-bundle (https://github.com/JCSDA-internal/soca/blob/6ef8be97e9d2be63e41c0fd8daedd78ea3eccc2f/bundle/CMakeLists.txt#L31) and turned on in the CI. I just haven't had time to look into it.
